src/unit_tests/variant/variant_cast_test.cpp

```diff
@@ -0,0 +1,241 @@
+/************************************************************************************
+*                                                                                   *
+*   Copyright (c) 2014, 2015 - 2017 Axel Menzel <info@rttr.org>                     *
+*                                                                                   *
+*   This file is part of RTTR (Run Time Type Reflection)                            *
+*   License: MIT License                                                            *
+*                                                                                   *
+*   Permission is hereby granted, free of charge, to any person obtaining           *
+*   a copy of this software and associated documentation files (the "Software"),    *
+*   to deal in the Software without restriction, including without limitation       *
+*   the rights to use, copy, modify, merge, publish, distribute, sublicense,        *
+*   and/or sell copies of the Software, and to permit persons to whom the           *
+*   Software is furnished to do so, subject to the following conditions:            *
+*                                                                                   *
+*   The above copyright notice and this permission notice shall be included in      *
+*   all copies or substantial portions of the Software.                             *
+*                                                                                   *
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      *
+*   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        *
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     *
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          *
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   *
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   *
+*   SOFTWARE.                                                                       *
+*                                                                                   *
+*************************************************************************************/
+
+#include <catch/catch.hpp>
+#include <iostream>
+#include <rttr/type>
+
+using namespace rttr;
+
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(variant*)", "[variant]")
+{
+    SECTION("empty cast")
+    {
+        variant var;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value == nullptr);
+    }
+
+    SECTION("valid cast")
+    {
+        variant var = 12;
+
+        auto value = variant_cast<int>(&var);
```

Comment:
Is that correct: value is still bound to the `var`. meaning if var gets destroyed, `value` is dangling as well.

Reply:
yes, that is correct. Same as with `variant::get_value<T>()`